libs/barista-components/drawer-table/src/drawer-table.ts

```diff
@@ -0,0 +1,193 @@
+/**
+ * @license
+ * Copyright 2020 Dynatrace LLC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  AfterContentInit,
+  AfterViewInit,
+  ChangeDetectionStrategy,
+  ChangeDetectorRef,
+  Component,
+  ContentChild,
+  Directive,
+  Input,
+  TemplateRef,
+  ViewChild,
+  ViewContainerRef,
+  ViewEncapsulation,
+} from '@angular/core';
+import {
+  DtHeaderRowDef,
+  DtRowDef,
+  DtTable,
+} from '@dynatrace/barista-components/table';
+
+import { DtDrawer } from '@dynatrace/barista-components/drawer';
+import { isEmpty } from '@dynatrace/barista-components/core';
+
+@Directive({
+  selector: '[dtDrawerContentRef]',
+})
+export class DtDrawerContentRef {}
+
+@Component({
+  selector: 'dt-drawer-table',
+  templateUrl: './drawer-table.html',
+  styleUrls: ['./drawer-table.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  encapsulation: ViewEncapsulation.Emulated,
+})
+export class DtDrawerTable<T> implements AfterViewInit, AfterContentInit {
+  /** Columns that stay visible when row is expanded  */
+  @Input() openColumns: string[];
+
+  /** Reference to the drawer element  */
+  @ViewChild(DtDrawer, { static: true }) private drawer: DtDrawer;
+
+  /** Reference to the drawer container  */
+  @ViewChild('container', { read: ViewContainerRef, static: false })
+  private container: ViewContainerRef;
+
+  /** @internal Template reference for the drawer content. */
+  @ContentChild(DtDrawerContentRef, { static: false, read: TemplateRef })
+  private _drawerContentRef: TemplateRef<{
+    $implicit: T;
+  }>;
+
+  /** @internal Reference to the table that is used to make the rows interactive */
+  @ContentChild(DtTable, { static: false }) private _table: DtTable<T>;
+
+  /** @internal Reference to the header row definition which is used to set the columns  */
+  @ContentChild(DtHeaderRowDef, { static: false })
+  private _rowHeaderDef: DtHeaderRowDef;
+
+  /** Reference to the row definition which is used to get and set the columns  */
+  @ContentChild(DtRowDef, { static: false }) rowDef: DtRowDef<T>;
+  private initialColumns: Iterable<string>;
```

Comment:
Prefix as internal with  _
```suggestion
  private _initialColumns: Iterable<string>;
```